### PR TITLE
RTFD: building docs for with redirect

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

ref: https://www.zonca.dev/posts/2020-09-01-redirect-readthedocs.html
it will apply only for `latest`

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @akihironitta

### CODE REVIEW

**Comment by Borda** [comment]
Instead of filtering, let's nuke it with echo the one index.rst

**Review by akihironitta — APPROVED** [review]

File: docs/source-pytorch/_index.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.. raw:: html
+
+    <script type="text/javascript">
+        window.location.replace('https://myorganization.github.io/myrepo/master/');
```
Comment (awaelchli) [inline docs/source-pytorch/_index.rst:4]: Doesn't this need to be filled in with the real URL?

File: .readthedocs.yml [hunk 2]
```diff
@@ -19,38 +19,19 @@
...
 
-submodules:
-  include: all
-  recursive: true
-
 # Build documentation in the docs/ directory with Sphinx
 # reference: https://docs.readthedocs.io/en/stable/config-file/v2.html#sphinx
 sphinx:
-  configuration: docs/source-pytorch/conf.py
+  configuration: docs/source-redirect/conf.py
```
Comment (Borda) [inline .readthedocs.yml:25]:
```suggestion
```

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
Perhaps these new files should be added to checkgroup as well? Not sure

**Pull request merged**